src/extractor/guidance/sliproad_handler.cpp

```diff
@@ -29,273 +32,730 @@ SliproadHandler::SliproadHandler(const IntersectionGenerator &intersection_gener
     {
-        const auto &next_data = node_based_graph.GetEdgeData(next_road.eid);
+        const auto &source_edge_data = node_based_graph.GetEdgeData(source_edge_id);
+        const auto &main_road_data = node_based_graph.GetEdgeData(main_road.eid);
+
+        const auto same_name = source_edge_data.name_id != EMPTY_NAMEID && //
+                               main_road_data.name_id != EMPTY_NAMEID &&   //
+                               !util::guidance::requiresNameAnnounced(source_edge_data.name_id,
+                                                                      main_road_data.name_id,
+                                                                      name_table,
+                                                                      street_name_suffix_table); //
 
-        const auto same_name =
-            next_data.name_id != EMPTY_NAMEID && source_edge_data.name_id != EMPTY_NAMEID &&
-            !util::guidance::requiresNameAnnounced(
-                next_data.name_id, source_edge_data.name_id, name_table, street_name_suffix_table);
+        using osrm::util::guidance::getTurnDirection;
 
         if (same_name)
         {
-            if (angularDeviation(next_road.angle, STRAIGHT_ANGLE) < 5)
-                intersection[obvious_turn_index].instruction.type = TurnType::Suppressed;
+            if (angularDeviation(main_road.angle, STRAIGHT_ANGLE) < 5)
+                intersection[*obvious].instruction.type = TurnType::Suppressed;
             else
-                intersection[obvious_turn_index].instruction.type = TurnType::Continue;
-            intersection[obvious_turn_index].instruction.direction_modifier =
-                getTurnDirection(intersection[obvious_turn_index].angle);
+                intersection[*obvious].instruction.type = TurnType::Continue;
+            intersection[*obvious].instruction.direction_modifier =
+                getTurnDirection(intersection[*obvious].angle);
         }
-        else if (next_data.name_id != EMPTY_NAMEID)
+        else if (main_road_data.name_id != EMPTY_NAMEID)
         {
-            intersection[obvious_turn_index].instruction.type = TurnType::NewName;
-            intersection[obvious_turn_index].instruction.direction_modifier =
-                getTurnDirection(intersection[obvious_turn_index].angle);
+            intersection[*obvious].instruction.type = TurnType::NewName;
+            intersection[*obvious].instruction.direction_modifier =
+                getTurnDirection(intersection[*obvious].angle);
         }
         else
         {
-            intersection[obvious_turn_index].instruction.type = TurnType::Suppressed;
+            intersection[*obvious].instruction.type = TurnType::Suppressed;
         }
     }
 
+    if (sliproad_found)
+    {
+        geojson_points.Write(std::vector<NodeID>{b, c, d});
+        geojson_lines.Write(std::vector<NodeID>{b, d});
+    }
+
     return intersection;
 }
 
+// Implementation details
+
```

Comment:
The interface (fun decls in the header) should have all documentation the callsite has to know about. I think.